DOCUMENTATION.md

```diff
@@ -0,0 +1,305 @@
+
+**Example:**
+
+```JavaScript
+var rootElement = document.getElementById( "impress" );
+rootElement.addEventListener( "impress:init", function() {
+  console.log( "Impress init" );
+});
+impress().init();
+```
+
+### impress().next()
+
+Navigates to the next step of the presentation using the [`goto()` function](#impressgotostepindexstepelementidstepelement-duration).
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.next();
+```
+
+### impress().prev()
+
+Navigates to the previous step of the presentation using the [`goto()` function](#impressgotostepindexstepelementidstepelement-duration).
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.prev();
+```
+
+### impress().goto(stepIndex|stepElementId|stepElement, [duration])
+
+Accepts a [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) that represents the step index.
+
+Navigates to the step given the provided step index.
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.goto(7);
+```
+
+Accepts a [`String`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String) that represents the [Step Element](#step-element) id.
+
+Navigates to the step given the provided [Step Element](#step-element) id.
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.goto( "overview" );
+```
+
+Accepts an [`HTMLElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement) that represents the [Step Element](#step-element).
+
+Navigates to the step given the provided [Step Element](#step-element).
+
+**Example:**
+
+```JavaScript
+var overview = document.getElementById( "overview" );
+var api = impress();
+api.init();
+api.goto( overview );
+```
+
+Accepts an optional [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) in the last argument that represents the duration of the transition in milliseconds. If not provided, the default transition duration for the presentation will be used.
+
+Triggers the `impress:stepenter` event in the [Root Element](#root-element) when the presentation navigates to the target [Step Element](#step-element).
+
+**Example:**
+
+```JavaScript
+var rootElement = document.getElementById( "impress" );
+rootElement.addEventListener( "impress:stepenter", function() {
+  var currentStep = document.querySelector( ".present" );
+  console.log( "Entered the Step Element '" + currentStep.id + "'" );
+});
+```
+
+Triggers the `impress:stepleave` event in the [Root Element](#root-element) when the presentation navigates away from the current [Step Element](#step-element).
+
+**Example:**
+```JavaScript
+var rootElement = document.getElementById( "impress" );
+rootElement.addEventListener( "impress:stepleave", function() {
+  console.log( "Left the current Step Element" );
```

Comment:
I guess it might be a good idea for the future, but for now it might be better documenting only what actually exists without focusing in additional things. Is there any way, even with step classes, to be able to retrieve the current step from the `impress:stepleave` event?